pkgs/development/python-modules/pydot_ng/default.nix

```
@@ -16,13 +17,17 @@ buildPythonPackage rec {
     sha256 = "8c8073b97aa7030c28118961e2c6c92f046e4cb57aeba7df87146f7baa6530c5";
   };
 
-  buildInputs = [ pytest unittest2 ];
+  patches = [ ./0001-graphviz_path.patch ];
+
+  postPatch =
+    ''
+    sed -i 's#NIX_GRAPHVIZ_PATH#"${pkgs.graphviz}/bin"#g' pydot_ng/__init__.py;
+    '';
+
+  buildInputs = [ pytest unittest2 mock ];
   propagatedBuildInputs = [ pkgs.graphviz pyparsing ];
 
-  checkPhase = ''
-    mkdir test/my_tests
-    py.test test
-  '';
+  checkPhase = "${pytest}/bin/py.test test";
```

It wasn't being found because pytest needs to be a checkInput.
```suggestion
  checkPhase = "py.test test";
```